### PR TITLE
update golangci config

### PR DESCRIPTION
**Pull request by srivickynesh**
update golangci config to include cache and additional linters support
fixed a couple of issues failing with linter

### CODE REVIEW

File: .tekton/go-lint-pipelinerun.yaml [hunk 1]
```diff
@@ -48,10 +49,17 @@ spec:
               script: | 
                 #!/usr/bin/env bash
                 cd $(workspaces.source.path)
+                export GOLANGCI_LINT_CACHE=$(workspaces.cache.path)/golangci-lint-cache
+                mkdir -p $GOLANGCI_LINT_CACHE
                 golangci-lint run -v --timeout=5m
+                cd $(workspaces.cache.path)/golangci-lint-cache
+                ls
+                cat *
```
Comment (ppitonak) [inline .tekton/go-lint-pipelinerun.yaml:57]: don't forget to remove before merging :)

File: pkg/triggers/helper.go [hunk 2]
```diff
@@ -2,7 +2,7 @@ package triggers
 
 import (
 	"crypto/hmac"
-	"crypto/sha1"
```
Comment (ppitonak) [inline pkg/triggers/helper.go:5]: Did you verify that test runs successfully?

File: pkg/k8s/k8s.go [hunk 3]
```diff
@@ -76,7 +76,8 @@ func WaitForServiceAccount(cs *clients.Clients, ns, targetSA string) *corev1.Ser
 		saList, err := cs.KubeClient.Kube.CoreV1().ServiceAccounts(ns).List(cs.Ctx, metav1.ListOptions{})
 		for _, sa := range saList.Items {
 			if sa.Name == targetSA {
-				ret = &sa
+				saCopy := sa
+				ret = &saCopy
```
[the remark this answers is not in the document]
Comment (srivickynesh) [reply pkg/k8s/k8s.go:80]: `Implicit Memory aliasing in for loop` pointer variable may reuse last item processed by for loop.

File: .tekton/go-lint-pipelinerun.yaml [hunk 4]
```diff
@@ -48,10 +49,14 @@ spec:
               script: | 
                 #!/usr/bin/env bash
                 cd $(workspaces.source.path)
+                export GOLANGCI_LINT_CACHE=$(workspaces.cache.path)/golangci-lint-cache
+                mkdir -p $GOLANGCI_LINT_CACHE
```
[the remark this answers is not in the document]
Comment (srivickynesh) [reply .tekton/go-lint-pipelinerun.yaml:53]: will do

File: pkg/triggers/helper.go (hunk 2 above)
Comment (srivickynesh) [reply pkg/triggers/helper.go:5]: I ran `gauge run --log-level=debug --verbose --tags 'e2e & !tls' specs/triggers` and test completed successfully.

**Review by ppitonak — APPROVED** [review]

**Pull request merged**